examples/vision/image_classification_efficientnet_fine_tuning.py

```diff
@@ -0,0 +1,423 @@
+callback). On CIFAR-100 with EfficientNetB0, this step will take validation accuracy to
+~70% with suitable (but not absolutely optimal) image augmentation. For this stage, using
+EfficientNetB0, validation accuracy and loss will be consistently better than training
+accuracy and loss. This is because the regularization is strong, which only
+suppresses train time metrics.
+
+Note that the convergence may take up to 50 epochs. If no data augmentation layer is
+applied, expect the validation accuracy to reach only ~60% even for many epochs.
+"""
+
+from tensorflow.keras.callbacks import ReduceLROnPlateau
+
+with strategy.scope():
+    model = build_model(n_classes=NUM_CLASSES)
+
+reduce_lr = ReduceLROnPlateau(
+    monitor="val_loss", factor=0.2, patience=5, min_lr=0.0001, verbose=2
+)
+
+epochs = 25  # @param {type: "slider", min:8, max:80}
+hist = model.fit(
+    ds_train, epochs=epochs, validation_data=ds_test, callbacks=[reduce_lr], verbose=2,
+)
+plot_hist(hist)
+
+"""
+The second step is to unfreeze a number of layers. Unfreezing layers and fine tuning is
+usually thought to only provide incremental improvements on validation accuracy, but for
+the case of EfficientNetB0 it boosts validation accuracy by about 10% to pass 80%
+(reaching ~87% as in the original paper requires including AutoAugmentation or Random
+Augmentaion).
+
+Note that the convergence may take more than 50 epochs. If no data augmentation layer is
+applied, expect the validation accuracy to reach only ~70% even for many epochs.
+"""
+
+
+def unfreeze_model(model):
+    for l in model.layers:
+        if "bn" in l.name:
+            print(f"{l.name} is staying untrainable")
+        else:
+            l.trainable = True
+
+    sgd = SGD(learning_rate=0.005)
+    model.compile(optimizer=sgd, loss="categorical_crossentropy", metrics=["accuracy"])
+    return model
+
+
+model = unfreeze_model(model)
+
+reduce_lr = ReduceLROnPlateau(
+    monitor="val_loss", factor=0.2, patience=5, min_lr=0.00001, verbose=2
+)
+epochs = 25  # @param {type: "slider", min:8, max:80}
+hist = model.fit(
+    ds_train, epochs=epochs, validation_data=ds_test, callbacks=[reduce_lr], verbose=2,
+)
+plot_hist(hist)
+
+"""
+### tips for fine tuning EfficientNet
+
+On unfreezing layers:
+- The batch normalization layers need to be kept untrainable
+(https://keras.io/guides/transfer_learning/). If they are also turned to trainable, the
+first epoch after unfreezing will significantly reduce accuracy.
+- In some cases it may be beneficial to open up only a portion of layers instead of
+unfreezing all. This will make fine tuning much faster when going to larger models like
+B7.
+- Each block needs to be all turned on or off. This is because the architecture includes
+a shortcut from the first layer to the last layer for each block. Not respecting blocks
+also significantly harms the final performance.
+
+
+Some other tips for utilizing EfficientNet
```

Comment:
Make sure to have a blank line before a list to ensure proper rendering of the list